### DISCUSSION TITLE
Scroll Restoration / History API

### DISCUSSION THREAD

**Discussion by Abiti-936** [solidjs-community/solid-primitives#96] [Ideas]
First of All, I would like to say Thanks for an Awesome Work in solid-app-router & it's integration with Solid Start.

I have seen Scroll Restoration or History API in Vue, Remix, Next.js, but Coming to Solid Start & solid-app-router I can't get it working. Every time when routing happens the page losts the Scroll Position.

Can You make Scroll Restoration in Solid  App Router?

**Comment by thetarnav** [comment]
solid-primitives is a project for primitives that are un-opinionated of the router solution. So you might have opened a discussion in the wrong repo. See [solid-app-router](https://github.com/solidjs/solid-app-router).

But if this is something that can be done in a router agnostic way, just by using the Browser API, then we could probably add it to the list.

**Comment by atk** [comment]
We could add an endpoint to the scroll primitives that returns `restorePos() & { recordPos: () => void }` for all the nodes (fallback: body) that are given with an extra option to auto-track the scroll position(s). That should it make fairly simple so create a solution on top of solid-app-router.

**Comment by Abiti-936** [comment]
Sorry I am looking for solid-app-router repo, it's my mistake.

https://github.com/solidjs/solid-app-router/issues/99
